### PR TITLE
Modularize NotEqual.

### PR DESCRIPTION
**Pull request by lina128**
Modularized NotEqual and refactored the kernel utils. This PR has below changes:

1. Refactored Div and SquaredDifferences to export two things: config and kernelFunc. Exposing kernelFunc allows kernels to call kernels just through kernelFunc.
2. Added an optional dtype to pass into binaryKernelFunc. This allows result to be created with this dtype as oppose to input's dtype. This is mainly used for comparison ops, such as Equal, NotEqual, Less, Greater, etc.
3. Modularized NotEqual.

---
This change is [Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3919)

### CODE REVIEW

File: tfjs-backend-cpu/src/utils/kernel_utils.ts [hunk 1]
```diff
@@ -15,37 +15,44 @@
...
-      const aVals = cpuBackend.data.get(a.dataId).values as TypedArray;
-      const bVals = cpuBackend.data.get(b.dataId).values as TypedArray;
-
-      const [resultData, resultShape] =
-          op(a.shape, b.shape, aVals, bVals, a.dtype);
-
-      const dataId = cpuBackend.write(resultData, resultShape, a.dtype);
-      return {dataId, shape: resultShape, dtype: a.dtype};
-    }
+export type SimpleBinaryOperation = (a: number, b: number) => number;
```
Comment (jinjingforever) [inline tfjs-backend-cpu/src/utils/kernel_utils.ts:26]: nit: might be useful to use this type to replace the type of the "op" parameter in createBinaryKernelImpl below.

**Review by jinjingforever — APPROVED** [review]

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3919)* status: :shipit: complete! 1 of 1 approvals obtained (waiting on @annxingyuan, @jinjingforever, and @tafsiri)

___
*[tfjs-backend-cpu/src/utils/kernel_utils.ts, line 26 at r1](https://reviewable.io/reviews/tensorflow/tfjs/3919#-MHDJSecISh6R8s3CScG-r1-26:-MHDTO-E-4wxeUqaQqZ7:bv5ddmq) ([raw file](https://github.com/tensorflow/tfjs/blob/0286d9d59b572c3d8f587f9b4658aa929750fb66/tfjs-backend-cpu/src/utils/kernel_utils.ts#L26)):*
<details><summary><i>Previously, jinjingforever (Jing Jin) wrote…</i></summary><blockquote>

nit: might be useful to use this type to replace the type of the "op" parameter in createBinaryKernelImpl below.
</blockquote></details>

Ah, good catch. Done.

**Review by tafsiri — APPROVED** [review]
LGTM with one suggestion. 

> Reviewed 4 of 5 files at r1, 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3919#-:-MHD_Fna1M5s4CBcdSof:b8xpi5f)* status: :shipit: complete! 2 of 1 approvals obtained (waiting on @annxingyuan, @jinjingforever, @lina128, and @tafsiri)

___
*[tfjs-backend-cpu/src/kernels/NotEqual.ts, line 22 at r2](https://reviewable.io/reviews/tensorflow/tfjs/3919#-MHDV2MJET6GTYfxiHXb:-MHDV2MJET6GTYfxiHXc:bvqkig4) ([raw file](https://github.com/tensorflow/tfjs/blob/e73b89e6193aaf40706d7d80ddd3588c6b1e94f8/tfjs-backend-cpu/src/kernels/NotEqual.ts#L22)):*
> ```
> createBinaryKernelConfig
> ```

More of a design/dx question. Could you pass the anonymous function directly to binaryKernelFunc below and have it call createBinaryKernelImpl internally? It seems like it is always called the same way by each caller so might as well be included in binaryKernelFunc (this would also affect your other pr with complex op examples). There could be a use case I am missing though.

**Review by annxingyuan — APPROVED** [review]
LGTM

> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3919#-:-MHGdc_J2G2c23V5yIWI:bb6ywmx)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @annxingyuan, @jinjingforever, and @lina128)

**Review by lina128 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/tensorflow/tfjs/3919)* status: :shipit: complete! 3 of 1 approvals obtained (waiting on @jinjingforever and @tafsiri)

___
*[tfjs-backend-cpu/src/kernels/NotEqual.ts, line 22 at r2](https://reviewable.io/reviews/tensorflow/tfjs/3919#-MHDV2MJET6GTYfxiHXb:-MHKU1_kEBK6ETp-Pn1j:bj9l15t) ([raw file](https://github.com/tensorflow/tfjs/blob/e73b89e6193aaf40706d7d80ddd3588c6b1e94f8/tfjs-backend-cpu/src/kernels/NotEqual.ts#L22)):*
<details><summary><i>Previously, tafsiri (Yannick Assogba) wrote…</i></summary><blockquote>

> ```
> createBinaryKernelConfig
> ```

More of a design/dx question. Could you pass the anonymous function directly to binaryKernelFunc below and have it call createBinaryKernelImpl internally? It seems like it is always called the same way by each caller so might as well be included in binaryKernelFunc (this would also affect your other pr with complex op examples). There could be a use case I am missing though. 
</blockquote></details>

Ah, good idea. Refactored.

**Pull request merged**